scripts/deploy-dockerhub.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# THIS IS MEANT TO BE RUN BY CI
+
+set -e
+
+# Usage: retry MAX CMD...
+# Retry CMD up to MAX times. If it fails MAX times, returns failure.
+# Example: retry 3 docker push "$DOCKERHUB_REPO:$TAG"
+function retry() {
+    max=$1
+    shift
+    count=1
+    until "$@"; do
+        count=$((count + 1))
+        if [[ $count -gt $max ]]; then
+            return 1
+        fi
+        echo "$count / $max"
+    done
+    return 0
+}
+
+# configure docker creds
+retry 3  echo "$DOCKER_PASS" | docker login -u="$DOCKER_USER" --password-stdin
```

Comment:
This doesn't seem right. I think you could do `retry 3 sh -c "echo \"$DOCKER_PASS\" | docker login -u=\"$DOCKER_USER\" --password-stdin`. There might also be a better way, although I'm not sure exactly what it is. TBH every single shell script I have ever written, I have eventually regretted writing as a shell script. Why not just write this thing in Python? It's not that complicated.

Also is this really supposed to be `-u=`? This seems like a gross violation of Unix CLI conventions.

Reply:
I copy/pasted this from another circle config from a different project.  This line is weird enough that I'm gonna guess somebody took some hits to get it right. :)

Reply:
Well, they didn't. I guess the punchline here is that `docker login` probably doesn't fail that often.

Try running this file. What should the output be?

```sh
#!/bin/bash

set -e

function retry() {
    max=$1
    shift
    count=1
    echo "Executing" "$@"
    until "$@"; do
        count=$((count + 1))
        if [[ $count -gt $max ]]; then
            return 1
        fi
        echo "$count / $max"
    done
    return 0
}

retry 3  true | false
```

Reply:
https://github.com/mozilla-services/kinto-dist/pull/436 :)